lib/wallet/src/Cardano/Wallet/Types/Read/Eras.hs

```diff
@@ -0,0 +1,94 @@
+
+-- |
+-- Copyright: © 2020-2022 IOHK
+-- License: Apache-2.0
+--
+-- Re-export `EraValue` library.
+--
+
+module Cardano.Wallet.Types.Read.Eras
+  ( -- * eras
+    KnownEras
+  , knownEraIndices
+    -- * era bounded values
+  , EraValue
+  , eraValueSerialize
+  , eraValueS
+  , fromEraValueS
+  , toEraValueS
```

Comment:
I would drop the `*S` family of things — we already have the following combinations to achieve the same effects:

* `extractEraValue` and `EraFunR`
* `inject` and `project`

Additionally, we also have `isoInAnyCardanoEra`, where `InAnyCardanoEra` duplicates the thing.

In order to keep cognitive load small, we have to stick to a style once we introduce it — either we predominantly use `EraFun`, then `case` and pattern matching is out; or we predominantly use `case` and pattern matching, then `EraFun` is out. 🤓

Reply:
Wait, these are all different things.
1) the isoInAnyCardanoEra is a conversion and  come with the price, it also changes slightly the semantics of the pattern matching 
2) EraFunR has nothing to do with `EraValue`, an EraValue is something in one era, an EraFun is a function on all eras
3) inject is about injecting  in an era and yes it superimpose with the EraValueS constructors
4) project looks good, but it is not, in fact it is producing a Maybe which means to extract a value you have to traverse quadratically because you do not know the era

So no, EraValueS is the best extractor we have now